### PR TITLE
fix: add step in confgen to handle network.tls[0] configs

### PR DESCRIPTION
**Pull request by jdogmcsteezy**

### CODE REVIEW

File: lib/utils/conf_gen.py [hunk 1]
```diff
@@ -457,10 +418,52 @@ async def __call__(self, context_dict: dict[str, Any]):
...
+               "name": "tls-name"
+            }
+        }
+    "network":
+        {
+            NamedSection("tls", "tls-name"): {
+               . . .
+            }
+        }
+    should be
```
Comment (dwelch-spike) [inline lib/utils/conf_gen.py:435]:
```suggestion
```
nit: is this needed?

File: lib/utils/conf_gen.py [hunk 2]
```diff
@@ -457,10 +418,52 @@ async def __call__(self, context_dict: dict[str, Any]):
             self._helper(config_dict)
 
 
-class ConvertIndexedToList(ConfigPipelineStep):
-    def __init__(self):
-        super().__init__()
+class ConvertIndexesSubcontextsToNamedSection(ConfigPipelineStep):
```
Comment (dwelch-spike) [inline lib/utils/conf_gen.py:421]:
```suggestion
class ConvertIndexedSubcontextsToNamedSection(ConfigPipelineStep):
```
nit: typo?

File: lib/utils/conf_gen.py [hunk 3]
```diff
@@ -457,10 +418,52 @@ async def __call__(self, context_dict: dict[str, Any]):
...
+
+                if isinstance(config, InterUnnamedSectionKey):
+                    config_split = config.type.split("[")
 
+                    if len(config_split) > 1:
+                        if "name" in config_dict[config]:
+                            name = config_dict[config]["name"]
+                            del config_dict[config]["name"]
+
+                            del config_dict[config]
```
Comment (dwelch-spike) [inline lib/utils/conf_gen.py:453]:
```suggestion
                            del config_dict[config]
```
I think only 1 delete is needed

**Review by dwelch-spike — COMMENTED** [review]
A few nitpicks. Also, is the pattern to unit test each class in the pipeline? If so this should have tests.

File: lib/utils/conf_gen.py (hunk 3 above)
Comment (jdogmcsteezy) [reply lib/utils/conf_gen.py:453]: It is needed. Ill rewrite it so it makes more sense though

**Review by dwelch-spike — COMMENTED** [review]
LGTM

**Pull request merged**